examples/Remarketing/AddFlightsFeed.php

```diff
@@ -0,0 +1,423 @@
+        ]);
+        $flightPriceMapping = new AttributeFieldMapping([
+            'feed_attribute_id' => new Int64Value([
+                'value' =>
+                    $placeHoldersToFeedAttributesMap[FlightPlaceholderField::FLIGHT_PRICE]
+                        ->getId()->getValue()
+            ]),
+            'flight_field' => FlightPlaceholderField::FLIGHT_PRICE
+        ]);
+        $flightSalePriceMapping = new AttributeFieldMapping([
+            'feed_attribute_id' => new Int64Value([
+                'value' =>
+                    $placeHoldersToFeedAttributesMap[FlightPlaceholderField::FLIGHT_SALE_PRICE]
+                        ->getId()->getValue()
+            ]),
+            'flight_field' => FlightPlaceholderField::FLIGHT_SALE_PRICE
+        ]);
+        $finalUrlsMapping = new AttributeFieldMapping([
+            'feed_attribute_id' => new Int64Value([
+                'value' =>
+                    $placeHoldersToFeedAttributesMap[FlightPlaceholderField::FINAL_URLS]
+                        ->getId()->getValue()
+            ]),
+            'flight_field' => FlightPlaceholderField::FINAL_URLS
+        ]);
+
+        // Creates the feed mapping.
+        $feedMapping = new FeedMapping([
+            'placeholder_type' => PlaceholderType::DYNAMIC_FLIGHT,
+            'feed' => new StringValue(['value' => $feedResourceName]),
+            'attribute_field_mappings' => [
+                $flightDescriptionMapping,
+                $destinationIdMapping,
+                $flightPriceMapping,
+                $flightSalePriceMapping,
+                $finalUrlsMapping
+            ]
+        ]);
+
+        // Creates the feed mapping operation.
+        $operation = new FeedMappingOperation();
+        $operation->setCreate($feedMapping);
+
+        // Issues a mutate request to add the feed mapping and print some information.
+        $feedMappingServiceClient = $googleAdsClient->getFeedMappingServiceClient();
+        $response = $feedMappingServiceClient->mutateFeedMappings($customerId, [$operation]);
+        printf(
+            "Feed mapping with resource name '%s' was created.%s",
+            $response->getResults()[0]->getResourceName(),
+            PHP_EOL
+        );
+    }
+
+    /**
+     * Adds a new item to the feed.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param string $feedResourceName the feed resource name for creating a feed item
+     * @param array $placeHoldersToFeedAttributesMap the map from placeholder fields to feed
+     *      attributes
+     */
+    private static function createFeedItem(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $feedResourceName,
+        array $placeHoldersToFeedAttributesMap
+    ) {
+        // Creates the flight description feed attribute value.
+        $flightDescriptionAttributeValue = new FeedItemAttributeValue([
```

Comment:
All these feed item attribute value creations could be factorized into a "createFeedItemAttributeValue" method directly called from the feedItem object creation

Reply:
Likewise, if you mean creating another private method, I'd rather not do it this time because it would cause some complication when we write a guide from these code examples. 
The current code, though not super efficient, I found the readability level is still reasonable.